SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplicationMultiworker.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestPropertyReplicationMultiworker.h"
+#include "Kismet/GameplayStatics.h"
+#include "ReplicatedTestActorBasic.h"
+
+/**
+ * "This tests that the data debug mode detects property changes on non-auth servers for basic types.
+ * This test contains 2 Servers and 2 Client workers.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *  - The authorative server spawns one ReplicatedTestActor
+ * - Test:
+ *  - All workers check that they can see exactly 1 ReplicatedTestActor.
+ *  - The authorative server changes the replicated properties.
+ *  - All workers check that the replicated properties have changed.
+ *  - The non-auth server changes the replicated properties which generates expected errors.
+ *  - Auth server check that the replicated properties are unchanged.
+ * - Clean-up:
+ *  - ReplicatedTestActor is destroyed using the RegisterAutoDestroyActor helper function.
+ */
+
+ASpatialTestPropertyReplicationMultiworker::ASpatialTestPropertyReplicationMultiworker()
+	: Super()
+{
+	Author = "Victoria Bloom";
+	Description = TEXT("This tests that the data debug mode detects property changes on non-auth servers for basic types.");
+}
+
+void ASpatialTestPropertyReplicationMultiworker::PrepareTest()
+{
+	Super::PrepareTest();
+
+	if (HasAuthority())
+	{
+		// Expected errors generated by changing replicated properties on a non-auth server
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was ReplicatedIntProperty!"),
+			1, true);
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was ReplicatedFloatProperty!"),
+			1, true);
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was bReplicatedBoolProperty!"),
+			1, true);
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was ReplicatedStringProperty!"),
+			1, true);
+	}
+
+	AddStep(
```

Comment:
Do we need to add steps to ensure the spawned actor is ready?

Reply:
In this case, I don't think so as we already wait for the TestActor to be Valid and then the steps themselves wait for the properties to be the expected values.